generators/app/templates/src/StarterKit/DataAccess/Options/DataAccessSettings.npg.cs

```diff
@@ -1,39 +1,56 @@
 using Microsoft.Extensions.DependencyInjection;
-using System;
 
 namespace StarterKit.DataAccess.Options
 {
-    public class DataAccessSettingsNpg
+  public class DataAccessSettingsNpg
+  {
+    public string Host { get; set; }
+    public string Port { get; set; }
+    public string DbName { get; set; }
+    public string User { get; set; }
+    public string Password { get; set; }
+
+    public static void RegisterConfiguration(IServiceCollection services, IConfigurationSection section)
+    {
+      services.Configure<DataAccessSettingsNpg>(settings =>
+      {
+        settings.LoadFromConfigSection(section);
+        settings.OverrideFromEnvironmentVariables();
+      });
+    }
+
+    public string GetConnectionString()
     {
-        public string Host { get; set; }
-        public string Port { get; set; }
-        public string DbName { get; set; }
-        public string User { get; set; }
-        public string Password { get; set; }
+      ushort port = 0;
+      try
+      {
+        port = ushort.Parse(Port);
+      }
+      catch (InvalidOperationException ex)
+      {
+        throw new InvalidOperationException("Database port must be a number from 0 to 65536.", ex.InnerException ?? ex);
+      }
 
-        public static void RegisterConfiguration(IServiceCollection services, IConfigurationSection section)
-        {
-            services.Configure<DataAccessSettingsNpg>(settings =>
-            {
-                settings.LoadFromConfigSection(section);
-                settings.OverrideFromEnvironmentVariables();
-            });
-        }
+      var connectionString = new ConnectionString(Host, port, DbName, User, Password);
+      return connectionString.ToString();
+    }
 
-        private void LoadFromConfigSection(IConfigurationSection section)
-        {
-            section.Bind(this);
-        }
+    private void LoadFromConfigSection(IConfigurationSection section)
+    {
+      section.Bind(this);
+    }
 
-        private void OverrideFromEnvironmentVariables()
-        {
-            var env = Environment.GetEnvironmentVariables();
-            Host = env.Contains("DB_POSTGRESQL_CONNECTION_HOST") ? env["DB_POSTGRESQL_CONNECTION_HOST"].ToString() : Host;
-            Port = env.Contains("DB_POSTGRESQL_CONNECTION_PORT") ? env["DB_POSTGRESQL_CONNECTION_PORT"].ToString() : Port;
-            DbName = env.Contains("DB_POSTGRESQL_CONNECTION_NAME") ? env["DB_POSTGRESQL_CONNECTION_NAME"].ToString() : DbName;
-            User = env.Contains("DB_POSTGRESQL_AUTH_USER") ? env["DB_POSTGRESQL_AUTH_USER"].ToString() : User;
-            Password = env.Contains("DB_POSTGRESQL_AUTH_PASSWORD") ? env["DB_POSTGRESQL_AUTH_PASSWORD"].ToString() : Password;
-        }
+    private void OverrideFromEnvironmentVariables()
+    {
+      var env = Environment.GetEnvironmentVariables();
+      Host = env.Contains("DB_POSTGRESQL_CONNECTION_HOST") ? env["DB_POSTGRESQL_CONNECTION_HOST"].ToString() : Host;
```

Comment:
De namen van deze configuratieparamters zouden ook als constanten kunnen gedefinieerd worden net zoals bij de AppSettingsConfigKeys

Reply:
Misschien best foutafhandeling toevoegen voor wanneer een parameter ontbreekt in AppConfig. Nu zou gewoon de setting uit de dataaccess.json file gebruikt worden wat niet de bedoeling is.
De container zal niet opgespind worden als er een setting ontbreekt.

if (env.IsDevelopment()) return;

           
            Host = Environment.GetEnvironmentVariable("DB_POSTGRESQL_CONNECTION_HOST");
            if (string.IsNullOrWhiteSpace(Host)) throw new ArgumentNullException("Configuration error: invalid parameter DB_POSTGRESQL_CONNECTION_HOST");